wagtail/admin/side_panels.py

```diff
@@ -96,13 +96,23 @@ class PageSidePanels:
     def __init__(self, request, page):
         self.request = request
         self.page = page
+        self.edit_handler = page.get_edit_handler()
+        self.form_class = self.edit_handler.get_form_class()
+        self.form = self.form_class(
+            instance=page,
+            for_user=request.user,
+        )
 
         self.side_panels = [
             StatusSidePanel(page, self.request),
-            CommentsSidePanel(page, self.request),
             # PreviewSidePanel(page),
         ]
 
+        if self.form.show_comments_toggle:
```

Comment:
Could we make this a keyword argument on the `__init__` method?

So change the above line to:

```python
def __init__(self, request, page, comments_enabled=False):
```

Then in the ``get_context_data`` method in the view, we can use the existing form instance:

```python
side_panels = PageSidePanels(self.request, self.page, comments_enabled=self.form.show_comments_toggle)
```